akka-typed/src/test/scala/akka/typed/ActorContextSpec.scala

```diff
@@ -0,0 +1,508 @@
+package akka.typed
+
+import scala.concurrent.duration._
+import scala.concurrent.Future
+import org.scalautils.ConversionCheckedTripleEquals
+import com.typesafe.config.ConfigFactory
+import akka.actor.DeadLetterSuppression
+
+object ActorContextSpec {
+  import ScalaDSL._
+
+  sealed trait Command
+  sealed trait Event
+
+  case class GotSignal(signal: Signal) extends Event with DeadLetterSuppression
+
+  case class Ping(replyTo: ActorRef[Pong]) extends Command
+  sealed trait Pong extends Event
+  case object Pong1 extends Pong
+  case object Pong2 extends Pong
+
+  case class Miss(replyTo: ActorRef[Missed.type]) extends Command
+  case object Missed extends Event
+
+  case class Renew(replyTo: ActorRef[Renewed.type]) extends Command
+  case object Renewed extends Event
+
+  case class Throw(ex: Exception) extends Command
+
+  case class MkChild(name: Option[String], monitor: ActorRef[GotSignal], replyTo: ActorRef[Created]) extends Command
+  case class Created(ref: ActorRef[Command]) extends Event
+
+  case class SetTimeout(duration: FiniteDuration, replyTo: ActorRef[TimeoutSet.type]) extends Command
+  case object TimeoutSet extends Event
+
+  case class Schedule[T](delay: FiniteDuration, target: ActorRef[T], msg: T, replyTo: ActorRef[Scheduled.type]) extends Command
+  case object Scheduled extends Event
+
+  case object Stop extends Command
+
+  case class Kill(name: String, replyTo: ActorRef[Killed.type]) extends Command
+  case object Killed extends Event
+
+  case class Watch(ref: ActorRef[Nothing], replyTo: ActorRef[Watched.type]) extends Command
+  case object Watched extends Event
+
+  case class Unwatch(ref: ActorRef[Nothing], replyTo: ActorRef[Unwatched.type]) extends Command
+  case object Unwatched extends Event
+
+  case class GetInfo(replyTo: ActorRef[Info]) extends Command
+  case class Info(self: ActorRef[Command], props: Props[Command], system: ActorSystem[Nothing]) extends Event
+
+  case class GetChild(name: String, replyTo: ActorRef[Child]) extends Command
+  case class Child(c: Option[ActorRef[Nothing]]) extends Event
+
+  case class GetChildren(replyTo: ActorRef[Children]) extends Command
+  case class Children(c: Set[ActorRef[Nothing]]) extends Event
+
+  case class ChildEvent(event: Event) extends Event
+
+  case class BecomeInert(replyTo: ActorRef[BecameInert.type]) extends Command
+  case object BecameInert extends Event
+
+  def subject(monitor: ActorRef[GotSignal]): Behavior[Command] =
```

Comment:
I could certainly be missing something, but it seems then that this function is effectively the new way to declare an Actor... an Actor being a combination of its ActorContext and Behaviour. Does that then mean that an Actor's behaviour must be a type hierarchy related to some parent type, in this case `Command`?